### PR TITLE
Feature/book controller crud

### PR DESCRIPTION
**Pull request by GabyCliff**
## Summary
- Created package exceptions
- Created class RestExceptionHandler wich handles custom exceptions
- Created class IdMismatchException, a custom RuntimeException
- Created class NotFoundException, a custom RuntimeException
- Added CRUD methods in BookController
## Trello Card
https://trello.com/c/BYKdkCVa/18-crear-las-operaciones-crud-de-book

### CODE REVIEW

File: src/main/java/com/wolox/training/exceptions/NotFoundException.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package com.wolox.training.exceptions;
+
+public class NotFoundException extends RuntimeException{
+    public NotFoundException() {
+    }
+
+    public NotFoundException(String message) {
+        super(message);
+    }
```
Comment (Jimenarosello) [inline src/main/java/com/wolox/training/exceptions/NotFoundException.java]: I think this method is not being used, you can remove it :)

**Review by Jimenarosello — APPROVED** [review]
Looks good to me! Nice job 😄 🚀

File: src/main/java/com/wolox/training/controllers/BookController.java [hunk 2]
```diff
@@ -1,16 +1,61 @@
...
+    }
+
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public Book create(@RequestBody Book book) {
+        return bookRepository.save(book);
+    }
+
+    @DeleteMapping("/{id}")
+    public void delete(@PathVariable Long id) {
```
Comment (rabinovichm) [inline src/main/java/com/wolox/training/controllers/BookController.java]: This is just extra information: you could add a response status with the NO_CONTENT code (204) to inform that the operation was done correctly and no content should be returned.

File: src/main/java/com/wolox/training/models/Book.java [hunk 3]
```diff
@@ -9,31 +13,40 @@
     @GeneratedValue(strategy = GenerationType.IDENTITY)
     private long id;
 
-    @Column(nullable = false)
+    @Column
+    @NotNull
```
Comment (rabinovichm) [inline src/main/java/com/wolox/training/models/Book.java:29]: The genre attribute is optional 😋

File: src/main/java/com/wolox/training/models/Book.java [hunk 4]
```diff
@@ -126,17 +143,12 @@ public void setIsbn(String isbn) {
 
     @Override
     public String toString() {
```
Comment (rabinovichm) [inline src/main/java/com/wolox/training/models/Book.java]: What do you need the `toString` method for? 🤔

File: src/main/java/com/wolox/training/exceptions/RestExceptionHandler.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+@ControllerAdvice
+public class RestExceptionHandler {
+
+    @ExceptionHandler(NotFoundException.class)
+    public ResponseEntity notFoundException (Exception e){
+        return ResponseEntity.status(HttpStatus.NOT_FOUND).body(e.getMessage());
```
Comment (rabinovichm) [inline src/main/java/com/wolox/training/exceptions/RestExceptionHandler.java]: I don't know if this is already happening (i can't download the code right now 😋), but you should inform the user that the thing that was not found was a `Book` and not a String for example. A message "book not found" or similar should appear somewhere. If this is already happening, please ignore this comment.

File: src/main/java/com/wolox/training/exceptions/RestExceptionHandler.java [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    }
+
+    @ExceptionHandler(IdMismatchException.class)
+    public ResponseEntity idMismatchException (Exception e){
+        return ResponseEntity.status(HttpStatus.CONFLICT).body(e.getMessage());
```
Comment (rabinovichm) [inline src/main/java/com/wolox/training/exceptions/RestExceptionHandler.java]: Same here, but with "book id mismatch".

**Review by rabinovichm — CHANGES_REQUESTED** [review]

File: src/main/java/com/wolox/training/exceptions/RestExceptionHandler.java (hunk 5 above)
Comment (GabyCliff) [reply src/main/java/com/wolox/training/exceptions/RestExceptionHandler.java]: Fixed in https://github.com/wolox-training/gf-java/pull/9

File: src/main/java/com/wolox/training/exceptions/RestExceptionHandler.java (hunk 6 above)
Comment (GabyCliff) [reply src/main/java/com/wolox/training/exceptions/RestExceptionHandler.java]: Fixed in https://github.com/wolox-training/gf-java/pull/9

**Review by rabinovichm — APPROVED** [review]

**Pull request merged**